cores/arduino/time.cpp

```diff
@@ -17,20 +19,25 @@ __attribute__((weak)) void yield() {
 }
 
 static FspTimer main_timer;
+// specifying these details as constants makes micros() faster !
+#define _timer_type                AGT_TIMER
+#define _timer_index               0
+#define _timer_get_underflow_bit() R_AGT0->AGTCR_b.TUNDF
+#define _timer_clock_divider       TIMER_SOURCE_DIV_8 // dividers 1, 2 and 8 work because _timer_period is < 16-bit. the divider 4 seems not supported: acts as 1
+#define _timer_clock_freq          24000000UL
```

Comment:
This might not work on all FSP supported boards (like Portenta C33). It's very important to fix this before merging since otherwise the number of ifdefs in the core would soon become unbearable

Reply:
Yes, the defines have to go @Kees-van-der-Oord , as already mentioned [here](https://github.com/arduino/ArduinoCore-renesas/pull/52#discussion_r1265004564).

I don't see how there can be any runtime speed-up by having them in the code, as its simple string search and replace done by the preprocessor.

Reply:
Please see the comment [here](https://github.com/arduino/ArduinoCore-renesas/pull/52#discussion_r1284799160). I will try to find the most efficient implementation that also works for the Portenta C33.